src/libappimage_shared/elf.c

```diff
@@ -1,3 +1,4 @@
+#define _FILE_OFFSET_BITS 64
```

Comment:
While doing some research on the topic anyway for zsync2/AppImageUpdate (which use libgpgme which in turn uses 64-bit types for file sizes apparently), I found that this is not the right way. You should set it as a compiler definition. Then, you can skip the detection logic and `mmap`/`open` work as expected.

In theory, we could also just hardcode `UINT_MAX`...